### PR TITLE
Move structure dict rendering from the renders into the bones

### PR DESCRIPTION
**Pull request by ArneGudermann**
This PR move the `renderBoneStructure` form the renderer to the  bones to eliminate multiple code. It also move `renderSkelStructure` in the `skeleton`. 
At this stage we still need to include the `render_type` to distinguish which renderer we are in.

### CODE REVIEW

**Comment by phorward** [comment]
@ArneGudermann can you please fix the conflicts that came in with the latest pull request merges first? After that I can start a review.

File: core/bones/base.py [hunk 1]
```diff
@@ -882,3 +882,40 @@ def iter_bone_value(
...
+        ret = {
+            "descr": str(self.descr),  # need to turn possible translate-object into string
+            "type": self.type,
+            "required": self.required,
+            "params": self.params,
+            "visible": self.visible,
+
+            "unique": self.unique.method.value if self.unique else False,
+            "languages": self.languages,
+            "emptyValue": self.getEmptyValue(),  # fixme: rename this into "emptyvalue", same as "defaultvalue"
```
Comment (phorward) [inline core/bones/base.py:900]:
```suggestion
            "emptyvalue": self.getEmptyValue(),  # fixme: rename this into "emptyvalue", same as "defaultvalue"
```

File: core/bones/base.py [hunk 2]
```diff
@@ -882,3 +882,40 @@ def iter_bone_value(
...
+        # Provide a defaultvalue, if it's not a function.
+        if not callable(self.defaultValue) and self.defaultValue is not None:
+            ret["defaultvalue"] = self.defaultValue
+
+        # Provide a multiple setting
+        if self.multiple and isinstance(self.multiple, MultipleConstraints):
+            ret["multiple"] = {
+                "minAmount": self.multiple.minAmount,
+                "maxAmount": self.multiple.maxAmount,
+                "preventDuplicates": self.multiple.preventDuplicates,
```
Comment (phorward) [inline core/bones/base.py:917]:
```suggestion
                "minamount": self.multiple.minAmount,
                "maxamount": self.multiple.maxAmount,
                "preventduplicates": self.multiple.preventDuplicates,
```

Comment (phorward) [reply core/bones/base.py:917]: We can do this, because they aren't evaluated somewhere, yet. (Not in the Vi!)

File: core/bones/spatial.py [hunk 3]
```diff
@@ -336,3 +336,9 @@ def setBoneValue(
             raise ValueError("append is not possible on %s bones" % self.type)
         assert isinstance(value, tuple) and len(value) == 2, "Value must be a tuple of (lat, lng)"
         skel[boneName] = value
+
+    def structure(self, render_type=None) -> dict:
+        return super().structure(render_type=render_type) | {
+            "boundsLat": self.boundsLat,
+            "boundsLng": self.boundsLng,
```
Comment (phorward) [inline core/bones/spatial.py:343]:
```suggestion
            "boundslat": self.boundsLat,
            "boundslng": self.boundsLng,
```

File: core/bones/string.py [hunk 4]
```diff
@@ -174,3 +174,8 @@ def getUniquePropertyIndexValues(self, skel, name: str) -> List[str]:
 
         return super().getUniquePropertyIndexValues(skel, name)
 
+    def structure(self, render_type=None) -> dict:
+        ret = super().structure(render_type=render_type) | {
+            "maxLength": self.maxLength
```
Comment (phorward) [inline core/bones/string.py:179]:
```suggestion
            "maxlength": self.maxLength
```

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann,
thanks for providing this pull request. 

Please check my suggestions, there is one error. 

Moreover, please entirely remove the flag `render_type`. Render-specific stuff should not be part of the bones!

I think, it's okay to make `readonly` the only valid key for all renders in ViUR-core 3.4. It is a breaking change and we should explicitly note on this, but one upper case "O" is not a reason to loop-through this parameter for only one case.

Furthermore, we should maybe think to generally write all keys in lower-case order. These changes are breaking, so a viur-core 3.4 must use a speciifc viur-vi 3.1 or similar, which is compatible.

File: core/bones/base.py [hunk 5]
```diff
@@ -882,3 +882,36 @@ def iter_bone_value(
...
+        ret = {
+            "descr": str(self.descr),  # need to turn possible translate-object into string
+            "type": self.type,
+            "required": self.required,
+            "params": self.params,
+            "visible": self.visible,
+            "readonly": self.readOnly,
+            "unique": self.unique.method.value if self.unique else False,
+            "languages": self.languages,
+            "emptyvalue": self.getEmptyValue(),  # fixme: rename this into "emptyvalue", same as "defaultvalue"
```
Comment (phorward) [inline core/bones/base.py:900]: The #fixme is now obsolete!
```suggestion
            "emptyvalue": self.getEmptyValue(),
```

File: core/bones/date.py [hunk 6]
```diff
@@ -254,3 +254,9 @@ def buildDBFilter(self,
     def performMagic(self, valuesCache, name, isAdd):
         if (self.creationMagic and isAdd) or self.updateMagic:
             valuesCache[name] = utcNow().replace(microsecond=0).astimezone(self.guessTimeZone())
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/date.py:258]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/file.py [hunk 7]
```diff
@@ -189,3 +189,8 @@ def recreateFileEntryIfNeeded(val):
                     return
                 importBlobFromViur2(val["dest"]["dlkey"], val["dest"]["name"])
                 recreateFileEntryIfNeeded(val["dest"])
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/file.py:193]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/record.py [hunk 8]
```diff
@@ -122,3 +122,8 @@ def getUniquePropertyIndexValues(self, valuesCache: dict, name: str) -> List[str
 
         """
         raise NotImplementedError
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/record.py:126]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/relational.py [hunk 9]
```diff
@@ -984,3 +984,12 @@ def getUniquePropertyIndexValues(self, valuesCache: dict, name: str) -> List[str
             return self._hashValueForUniquePropertyIndex(value["dest"]["key"])
         elif isinstance(value, list):
             return self._hashValueForUniquePropertyIndex([x["dest"]["key"] for x in value])
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/relational.py:988]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/select.py [hunk 10]
```diff
@@ -58,3 +58,8 @@ def singleValueFromClient(self, value, skel, name, origData):
                 return key, None
         return self.getEmptyValue(), [
             ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Invalid value selected")]
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/select.py:62]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/spatial.py [hunk 11]
```diff
@@ -336,3 +336,9 @@ def setBoneValue(
             raise ValueError("append is not possible on %s bones" % self.type)
         assert isinstance(value, tuple) and len(value) == 2, "Value must be a tuple of (lat, lng)"
         skel[boneName] = value
+
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/spatial.py:340]:
```suggestion
    def structure(self) -> dict:
```

File: core/bones/string.py [hunk 12]
```diff
@@ -174,3 +174,8 @@ def getUniquePropertyIndexValues(self, skel, name: str) -> List[str]:
 
         return super().getUniquePropertyIndexValues(skel, name)
 
+    def structure(self, ) -> dict:
```
Comment (phorward) [inline core/bones/string.py:177]:
```suggestion
    def structure(self) -> dict:
```

File: core/skeleton.py [hunk 13]
```diff
@@ -253,12 +253,17 @@ def setEntity(self, entity: db.Entity):
         self.accessedValues = {}
         self.renderAccessedValues = {}
 
+    def structure(self, ):
```
Comment (phorward) [inline core/skeleton.py:256]:
```suggestion
    def structure(self):
```

**Review by phorward — CHANGES_REQUESTED** [review]
This is much better. Can you please fix the suggestions?

**Comment by phorward** [comment]
I have one more proposal which we should discuss in the viur-meeting: In case we want a full backward compatibility during ViUR3, we can assign several keys twice, like `emptyvalue` and `emptyValue`. Always with a comment for deprecation and removal in ViUR4.

File: core/bones/base.py [hunk 14]
```diff
@@ -882,3 +882,36 @@ def iter_bone_value(
...
+
+        # Provide a defaultvalue, if it's not a function.
+        if not callable(self.defaultValue) and self.defaultValue is not None:
+            ret["defaultvalue"] = self.defaultValue
+
+        # Provide a multiple setting
+        if self.multiple and isinstance(self.multiple, MultipleConstraints):
+            ret["multiple"] = {
+                "minamount": self.multiple.minAmount,
+                "maxamount": self.multiple.maxAmount,
```
Comment (phorward) [inline core/bones/base.py:912]: I'm a fan of short names without redundant information. What do you think? 
```suggestion
                "min": self.multiple.minAmount,
                "max": self.multiple.maxAmount,
```

File: core/bones/file.py [hunk 15]
```diff
@@ -189,3 +189,8 @@ def recreateFileEntryIfNeeded(val):
                     return
                 importBlobFromViur2(val["dest"]["dlkey"], val["dest"]["name"])
                 recreateFileEntryIfNeeded(val["dest"])
+
+    def structure(self) -> dict:
+        return super().structure() | {
+            "validMimeTypes": self.validMimeTypes
```
Comment (sveneberth) [inline core/bones/file.py:195]: This is still camelCase?

Maybe replace by
```suggestion
            "valid_mime_types": self.validMimeTypes
```
or 

```suggestion
            "validmimetypes": self.validMimeTypes
```
?

File: core/bones/text.py [hunk 16]
```diff
@@ -361,3 +361,8 @@ def getUniquePropertyIndexValues(self, valuesCache: dict, name: str) -> List[str
             raise NotImplementedError()
 
         return super().getUniquePropertyIndexValues(valuesCache, name)
+
+    def structure(self) -> dict:
+        return super().structure() | {
+            "validHtml": self.validHtml,
```
Comment (sveneberth) [inline core/bones/text.py:367]: keep it camelCase?

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: core/bones/text.py (hunk 16 above)
Comment (phorward) [reply core/bones/text.py:367]: never!

**Review by sveneberth — DISMISSED** [review]

File: core/skeleton.py [hunk 17]
```diff
@@ -253,12 +253,17 @@ def setEntity(self, entity: db.Entity):
         self.accessedValues = {}
         self.renderAccessedValues = {}
 
+    def structure(self):
+        return [(key, bone.structure()) for key, bone in self.items()]
```
Comment (phorward) [inline core/skeleton.py:257]: Relates to #649

**Review by phorward — CHANGES_REQUESTED** [review]
Hello @ArneGudermann, this looks nice already.

I still have one thing I want to change before merging: `skel.structure()` should return a dict. The `[[key, descr]...]`-list notation is a specialty of the JSON-renderer, and should only be generated there.

**Comment by phorward** [comment]
@ArneGudermann if you approve #651 all my wishes are being respected! :1st_place_medal:

**Comment by phorward** [comment]
@sveneberth I've merged #651 into this PR to provide two flags to make this PR fully backward-compatible with `viur-core<3.4`, but to allow configure a project to use new features on demand. Can you please review this, so this can be merged, making a release candidate possible?

**Comment by sveneberth** [comment]
from the code it looks okay, but my Vi crashes:
```
['user', 'list']
15pyodide.asm.js:14 userlist does not exist or is not registered!
pyodide.asm.js:14 Python exception:
Traceback (most recent call last):
  File "/build/flare/flare/network.py", line 125, in onReadyStateChange
  File "/build/flare/flare/network.py", line 519, in onCompletion
  File "/build/widgets/list.py", line 370, in receivedStructure
ValueError: too many values to unpack (expected 2)


log_python_error @ pyodide.asm.js:14
$pythonexc2js @ pyodide.asm.wasm:0xe6fb3
Module._pythonexc2js @ pyodide.asm.js:14
Module.callPyObjectKwargs @ pyproxy.gen.js:405
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076
XMLHttpRequest.send (async)
hiwire_call_bound @ pyodide.asm.js:14
$func349 @ pyodide.asm.wasm:0xe87a1
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ed0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1236bd
$_pyproxy_apply @ pyodide.asm.wasm:0xea93b
Module.callPyObjectKwargs @ pyproxy.gen.js:391
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076
hiwire_call_bound @ pyodide.asm.js:14
$func349 @ pyodide.asm.wasm:0xe87a1
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ed0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f91
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x121463
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
$_PyObject_Call @ pyodide.asm.wasm:0x1219c9
$PyObject_Call @ pyodide.asm.wasm:0x121a2d
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f243a
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1237c9
$PyVectorcall_Call @ pyodide.asm.wasm:0x121865
$_PyObject_Call @ pyodide.asm.wasm:0x12195f
$PyObject_Call @ pyodide.asm.wasm:0x121a2d
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f243a
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1236bd
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_pyproxy_apply @ pyodide.asm.wasm:0xea93b
Module.callPyObjectKwargs @ pyproxy.gen.js:391
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076
pyodide.asm.js:14 Uncaught PythonError: Traceback (most recent call last):
  File "/build/flare/flare/network.py", line 125, in onReadyStateChange
  File "/build/flare/flare/network.py", line 519, in onCompletion
  File "/build/widgets/list.py", line 370, in receivedStructure
ValueError: too many values to unpack (expected 2)

    at new_error (pyodide.asm.js:14:247183)
    at pyodide.asm.wasm:0xe6eb0
    at pyodide.asm.wasm:0xe6fae
    at Module._pythonexc2js (pyodide.asm.js:14:944230)
    at Module.callPyObjectKwargs (pyproxy.gen.js:405:12)
    at Module.callPyObject (pyproxy.gen.js:411:17)
    at PyProxyClass.apply (pyproxy.gen.js:1199:19)
    at Object.apply (pyproxy.gen.js:1076:18)
new_error @ pyodide.asm.js:14
$wrap_exception @ pyodide.asm.wasm:0xe6eb0
$pythonexc2js @ pyodide.asm.wasm:0xe6fae
Module._pythonexc2js @ pyodide.asm.js:14
Module.callPyObjectKwargs @ pyproxy.gen.js:405
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076
XMLHttpRequest.send (async)
hiwire_call_bound @ pyodide.asm.js:14
$func349 @ pyodide.asm.wasm:0xe87a1
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ed0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1236bd
$_pyproxy_apply @ pyodide.asm.wasm:0xea93b
Module.callPyObjectKwargs @ pyproxy.gen.js:391
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076
hiwire_call_bound @ pyodide.asm.js:14
$func349 @ pyodide.asm.wasm:0xe87a1
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ed0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f91
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x121463
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
$_PyObject_Call @ pyodide.asm.wasm:0x1219c9
$PyObject_Call @ pyodide.asm.wasm:0x121a2d
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f243a
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_PyObject_FastCallDictTstate @ pyodide.asm.wasm:0x12142d
$_PyObject_Call_Prepend @ pyodide.asm.wasm:0x121ef6
$func2315 @ pyodide.asm.wasm:0x17f2a2
$func2220 @ pyodide.asm.wasm:0x176e7a
method_call_trampoline @ pyodide.asm.js:14
$_PyObject_MakeTpCall @ pyodide.asm.wasm:0x121318
$func3026 @ pyodide.asm.wasm:0x1f41e1
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1237c9
$PyVectorcall_Call @ pyodide.asm.wasm:0x121865
$_PyObject_Call @ pyodide.asm.wasm:0x12195f
$PyObject_Call @ pyodide.asm.wasm:0x121a2d
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f243a
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$func979 @ pyodide.asm.wasm:0x121c58
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121aa0
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1ef0
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$func1022 @ pyodide.asm.wasm:0x1236bd
$func3026 @ pyodide.asm.wasm:0x1f41fe
$_PyEval_EvalFrameDefault @ pyodide.asm.wasm:0x1f1f4d
$_PyEval_EvalCode @ pyodide.asm.wasm:0x1ed755
$_PyFunction_Vectorcall @ pyodide.asm.wasm:0x121b2f
$_pyproxy_apply @ pyodide.asm.wasm:0xea93b
Module.callPyObjectKwargs @ pyproxy.gen.js:391
Module.callPyObject @ pyproxy.gen.js:411
apply @ pyproxy.gen.js:1199
apply @ pyproxy.gen.js:1076

```


even if set `conf["viur.compatibility"] = []`

**Comment by phorward** [comment]
> So it looks like the compatibility isn't working...

Damn! You're right and I've fixed it also for the vi in [4c59569](https://github.com/viur-framework/viur-core/pull/637/commits/4c59569181098b413d0531059d29e8dccbb6ac06). Please retry.

**Review by phorward — APPROVED** [review]

**Review by sveneberth — APPROVED** [review]

**Pull request merged**